File: ios/PulseBoard/Domain/UseCases.swift
final class LoadUsersUseCase {
  private let repository: UserRepositoryProtocol
  init(repository: UserRepositoryProtocol) { self.repository = repository }
  func execute() -> [UserProfile] {
    repository.getUsers().sorted { $0.lastUpdated > $1.lastUpdated }
  }
}
final class LoadUserDetailUseCase {
  private let repository: UserRepositoryProtocol
  init(repository: UserRepositoryProtocol) { self.repository = repository }
  func execute(id: Int) -> UserProfile? { repository.getUser(id: id) }
}

File: android/app/src/main/java/com/belldien/pulseboard/domain/UseCases.kt
package com.belldien.pulseboard.domain

import com.belldien.pulseboard.data.UserRepository
import com.belldien.pulseboard.model.UserProfile

class LoadUsersUseCase(private val repository: UserRepository) {
    operator fun invoke(): List<UserProfile> =
        repository.getUsers().sortedByDescending { it.lastUpdated }
}

class LoadUserDetailUseCase(private val repository: UserRepository) {
    operator fun invoke(id: Int): UserProfile? = repository.getUser(id)
}
